File: src/backend/TrainingManagement.Api/Entities/Blacklist.cs
namespace TrainingManagement.Api.Entities;

// 黑名单实体，对应数据库BLACKLIST表
public sealed class Blacklist
{
    // 黑名单记录编号（主键，序列生成）
    public long BlackId { get; set; }

    // 员工编号（外键，关联EMPLOYEES）
    public long EmpId { get; set; }

    // 黑名单原因
    public string Reason { get; set; } = string.Empty;

    // 开始日期
    public DateTime StartDate { get; set; }

    // 结束日期
    public DateTime? EndDate { get; set; }

    // 黑名单状态：ACTIVE/RELEASED
    public string Status { get; set; } = "ACTIVE";

    // 操作人员工编号（审计字段，历史数据可为空）
    public long? OperatorEmpId { get; set; }

    // 创建时间
    public DateTime CreatedAt { get; set; }

    // 员工姓名（联表展示字段，非持久化）
    public string EmpName { get; set; } = string.Empty;

    // 员工部门名称（联表展示字段，非持久化）
    public string DeptName { get; set; } = string.Empty;
}

File: src/frontend/src/api/mappers/blacklist.ts
import { fromEnvelopeFailure } from '@/api/error'
import type { ApiEnvelopeDto, BlacklistDto, PageDto } from '@/api/transport'
import type { PageResult } from '@/types/api'
import type { BlacklistRecord, BlacklistStatus } from '@/domains/blacklist'
import { BLACKLIST_STATUS_LABELS } from '@/domains/blacklist'

const mapStatus = (value: string): BlacklistStatus =>
  value === 'ACTIVE' || value === 'RELEASED' ? value : 'UNKNOWN'

export function mapBlacklistRecord(dto: BlacklistDto): BlacklistRecord {
  const status = mapStatus(dto.status)
  return {
    id: String(dto.blackId),
    empId: dto.empId,
    empName: dto.empName,
    deptName: dto.deptName,
    reason: dto.reason,
    startDate: dto.startDate,
    endDate: dto.endDate,
    status,
    statusLabel: BLACKLIST_STATUS_LABELS[status],
    operatorEmpId: dto.operatorEmpId,
    createdAt: dto.createdAt,
  }
}

export function mapBlacklist(envelope: ApiEnvelopeDto<BlacklistDto>): BlacklistRecord {
  if (!envelope.success || !envelope.data) throw fromEnvelopeFailure(envelope)
  return mapBlacklistRecord(envelope.data)
}

export function mapBlacklistPage(
  envelope: ApiEnvelopeDto<PageDto<BlacklistDto>>,
): PageResult<BlacklistRecord> {
  if (!envelope.success || !envelope.data) throw fromEnvelopeFailure(envelope)
  return { ...envelope.data, items: envelope.data.items.map(mapBlacklistRecord) }
}
